guardrails/run/async_stream_runner.py

```diff
@@ -134,6 +139,8 @@ async def async_step(
                     "$",
                     "$",
                     True,
+                    context=context,
```

Comment:
Quick note on this: the right way to solve this would be to run the entire stream within a specific context and then we wouldn't/shouldn't need to pass this and run just the var ops with it.  The reason I didn't do that now is because that could interfere with telemetry contexts so we would need extensive regression testing.   We should still improve this in the future, but this patch can be released sooner with less impact.